apollo-gradle-plugin/src/main/kotlin/com/apollographql/apollo/gradle/internal/DefaultCompilationUnit.kt

```diff
@@ -45,17 +45,29 @@ abstract class DefaultCompilationUnit @Inject constructor(
     }
 
     if (!compilerParams.schemaFile.isPresent) {
-      compilerParams.schemaFile.set {
-        project.file(
+      // This needs to be lazy for:
+      // - test variants that don't have a schema
```

Comment:
This is a pain. For 3.0, I'd like to do something about test variants. Most likely remove them... Do we know if anyone uses that feature?

Reply:
Yeah we can remove this. not really sure what is the benefit of having unique graphql queries for test variants